lib/matplotlib/tests/test_ticker.py

```diff
@@ -482,82 +501,104 @@ def test_LogFormatterSciNotation():
+        assert '00002' == tmp_form(2)
+
+        # test str.format() style formatter with `pos`
+        tmp_form = mticker.StrMethodFormatter('{x:03d}-{pos:02d}')
+        assert '002-01' == tmp_form(2, 1)
+
+
+class TestPercentFormatter(object):
+    percent_data = [
+        # Check explicitly set decimals over different intervals and values
+        (100, 0, '%', 120, 100, '120%'),
+        (100, 0, '%', 100, 90, '100%'),
+        (100, 0, '%', 90, 50, '90%'),
+        (100, 0, '%', 1.7, 40, '2%'),
+        (100, 1, '%', 90.0, 100, '90.0%'),
+        (100, 1, '%', 80.1, 90, '80.1%'),
+        (100, 1, '%', 70.23, 50, '70.2%'),
+        # 60.554 instead of 60.55: see https://bugs.python.org/issue5118
+        (100, 1, '%', 60.554, 40, '60.6%'),
+        # Check auto decimals over different intervals and values
+        (100, None, '%', 95, 1, '95.00%'),
+        (1.0, None, '%', 3, 6, '300%'),
+        (17.0, None, '%', 1, 8.5, '6%'),
+        (17.0, None, '%', 1, 8.4, '5.9%'),
+        (5, None, '%', -100, 0.000001, '-2000.00000%'),
+        # Check percent symbol
+        (1.0, 2, None, 1.2, 100, '120.00'),
+        (75, 3, '', 50, 100, '66.667'),
+        (42, None, '^^Foobar$$', 21, 12, '50.0^^Foobar$$'),
+    ]
+
+    percent_ids = [
+        # Check explicitly set decimals over different intervals and values
+        'decimals=0, x>100%',
+        'decimals=0, x=100%',
+        'decimals=0, x<100%',
+        'decimals=0, x<0%',
+        'decimals=1, x>100%',
+        'decimals=1, x=100%',
+        'decimals=1, x<100%',
+        'decimals=1, x<0%',
+        # Check auto decimals over different intervals and values
+        'autodecimal, x<100%, display_range=1',
+        'autodecimal, x>100%, display_range=6 (custom xmax test)',
+        'autodecimal, x<100%, display_range=8.5 (autodecimal test 1)',
+        'autodecimal, x<100%, display_range=8.4 (autodecimal test 2)',
+        'autodecimal, x<-100%, display_range=1e-6 (tiny display range)',
+        # Check percent symbol
+        'None as percent symbol',
+        'Empty percent symbol',
+        'Custom percent symbol',
+    ]
+
+    @pytest.mark.parametrize(
+            'xmax, decimals, symbol, x, display_range, expected',
+            percent_data, ids=percent_ids)
+    def test_percentformatter(self, xmax, decimals, symbol,
+                              x, display_range, expected):
+        formatter = mticker.PercentFormatter(xmax, decimals, symbol)
+        assert formatter.format_pct(x, display_range) == expected
+
+
+class TestEngFormatter(object):
+    def test_formatting(self):
+        """
+        Create two instances of EngFormatter with default parameters, with and
+        without a unit string ('s' for seconds). Test the formatting in some
+        cases, especially the case when no SI prefix is present, for values in
+        [1, 1000).
+
+        Should not raise exceptions.
+        """
+        unitless = mticker.EngFormatter()
```

Comment:
these should probably be two separate tests (unitless and with unit) and possibly parameterized